### DISCUSSION TITLE
What limit value should I set to MagickSetResourceLimit, if I want to the ResourceLimit is unlimited?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#7119] [Development]
### ImageMagick version

7.1.0-52

### Operating system, version and so on

Debian 6.3.0-18+deb9u1

### Description

```
func (mw *MagickWand) SetResourceLimit(rtype ResourceType, limit int64) error {
	ok := C.MagickSetResourceLimit(C.ResourceType(rtype), C.MagickSizeType(limit))
	return mw.getLastErrorIfFailed(ok)
}
```

What limit value should I set to MagickSetResourceLimit, if I want to the ResourceLimit is unlimited?

**Comment by snibgo** [comment]
If you look at the source code, MagickSetResourceLimit calls SetMagickResourceLimit, which sets the new limit to the given limit if the given limit is more restrictive than the old limit. The function will not relax limits, and can't be used to change a limit to "unlimited".
